src/iarc7_motion/iarc_tasks/task_utilities/pid_controller.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+import rospy
+import numpy as np
+
+class PidSettings(object): 
+    def __init__(self, settings_dict):
+        self.kp = settings_dict.get('kp', 0)
```

Comment:
I don't think you want defaults here, you probably just want to use a normal dict lookup and let the `KeyError` get thrown if the setting isn't there.

Reply:
fixed